bind.go

```diff
@@ -144,11 +144,20 @@ func (b *DefaultBinder) bindData(destination interface{}, data map[string][]stri
 	for i := 0; i < typ.NumField(); i++ {
 		typeField := typ.Field(i)
 		structField := val.Field(i)
+		if typeField.Anonymous {
+			for structField.Kind() == reflect.Ptr {
+				structField = structField.Elem()
+			}
+		}
 		if !structField.CanSet() {
 			continue
 		}
 		structFieldKind := structField.Kind()
 		inputFieldName := typeField.Tag.Get(tag)
+		if typeField.Anonymous && structField.Kind() == reflect.Struct && inputFieldName != "" {
```

Comment:
I think this case should return an error and not silently hide the error that developer has made.
Let's just enforce rule that

This is NOT valid/allowed:
```go
result := struct {
		*Bar `query:"bar"`   // or param:"bar"
	}{&Bar{}}
```

This is OK:
```go
result := struct {
		*Bar  // or json:"bar" as json has its own implementation
	}{&Bar{}}
```